### ISSUE TITLE
[Feature Request] documentation of Delta configs / table properties missing (e.g. `checkpointInterval`, `enableDeletionVectors`, `enableExpiredLogCleanup`)

### ISSUE THREAD

**Issue by keen85** [delta-io/delta#1729]
### Overview

Official [Delta table properties reference](https://docs.delta.io/latest/table-properties.html#delta-table-properties-reference) is missing lots of [implemented delta configs](https://github.com/delta-io/delta/blob/master/core/src/main/scala/org/apache/spark/sql/delta/DeltaConfig.scala).
For example:
- `checkpointInterval`
- `enableExpiredLogCleanup`
- `enableDeletionVectors`
- `enableFullRetentionRollback`
- `enableRowIds`

I was wondering if it was not possible to **generate** the documentation from the documented source code **automatically**.

### Willingness to contribute

The Delta Lake Community encourages new feature contributions. Would you or another member of your organization be willing to contribute an implementation of this feature?

- [ ] Yes. I can contribute this feature independently.
- [ ] Yes. I would be willing to contribute this feature with guidance from the Delta Lake community.
- [x] No. I cannot contribute this feature at this time.

**Comment by allisonport-db** [comment]
The properties listed (except `enableDeletionVectors`) are internal properties that we intentionally don't document in the docs. We will add `enableDeletionVectors` for the 2.4 release.

**Comment by keen85** [comment]
Hi @allisonport-db ,
wouldn't be `checkpointInterval` also an Table Config that is interesting for end users?
I think e.g. for streaming use cases where there are very frequent appends, users might want to set `checkpointInterval` to a higher value in order to increase write performance?

**Comment by agrueneberg** [comment]
I agree with @keen85 that `checkpointInterval` should be documented. Coming from Trino, where it is documented, I was surprised to not find it in the table properties reference. Also, 2.4 has been released and `enableDeletionVectors` is still missing.

**Comment by foss-contributor** [comment]
@keen85, thank you for your contribution! This issue has been inactive for over 18 months and will be closed in one week. To keep it open, please leave a comment on this issue or contact us in the `#contributors` channel on [Delta Lake Slack](https://go.delta.io/slack). We appreciate your involvement in Delta Lake!

**Comment by felipepessoto** [comment]
Not stale
